app/lib/shared/markdown.dart

```diff
@@ -25,14 +25,42 @@ String markdownToHtml(String text, String baseUrl) {
   final lines = text.replaceAll('\r\n', '\n').split('\n');
   final nodes = document.parseLines(lines);
 
-  final visitor = new _UrlRewriter(sanitizedBaseUrl);
+  final urlRewriter = _UrlRewriter(sanitizedBaseUrl);
+  final hashLink = _HashLink();
   for (final node in nodes) {
-    node.accept(visitor);
+    node.accept(urlRewriter);
+    node.accept(hashLink);
   }
 
   return m.renderToHtml(nodes) + '\n';
 }
 
+/// Adds an extra <a href="#hash">#</a> element to all h1,h2,h3..h6 elements.
+class _HashLink implements m.NodeVisitor {
+  @override
+  void visitText(m.Text text) {}
+
+  @override
+  bool visitElementBefore(m.Element element) => true;
+
+  @override
+  void visitElementAfter(m.Element element) {
+    if (!element.tag.startsWith('h') ||
```

Comment:
Nit: I think this would read easier as:
```
if (!(element.tag.startsWith('h') &&
    element.tag.length == 2 &&
    element.generatedId != null &&
    element.children.length == 1)) {
  return;
}
```